### ISSUE TITLE
Incompatability with Project Factory

### ISSUE THREAD

**Issue by onetwopunch** [terraform-google-modules/terraform-google-bastion-host#19]
When passing in a project id generated from project factory, you get the following error: 
```
Error: Invalid for_each argument

  on .terraform/modules/bastion/terraform-google-modules-terraform-google-bastion-host-7d2e49c/main.tf line 88, in resource "google_project_iam_member" "bastion_sa_bindings":
  88:   for_each = toset(compact(concat(
  89:     var.service_account_roles,
  90:     var.service_account_roles_supplemental,
  91:     ["projects/${var.project}/roles/${google_project_iam_custom_role.compute_os_login_viewer.role_id}"]
  92:   )))
```
The "for_each" value depends on resource attributes that cannot be determined
until apply, so Terraform cannot predict how many instances will be created.
To work around this, use the -target argument to first apply only the
resources that the for_each depends on.

We should pull this interpolated line out of that for_each loop into its own resource to avoid this.

**Comment by morgante** [comment]
Wasn't this solved in #16?

**Comment by onetwopunch** [comment]
Yes I believe it was.
